plugins/sql-language/src/main/java/org/polypheny/db/sql/language/ddl/altertable/SqlAlterTableAddForeignKey.java

```diff
@@ -108,31 +97,19 @@ public void unparse( SqlWriter writer, int leftPrec, int rightPrec ) {
 
 
     @Override
-    public void execute( Context context, Statement statement, QueryParameters parameters ) {
-        CatalogTable catalogTable = getCatalogTable( context, table );
-        CatalogTable refTable = getCatalogTable( context, referencesTable );
-
-        // Make sure that this is a table of type TABLE (and not SOURCE)
-        if ( catalogTable.entityType != EntityType.ENTITY ) {
-            throw CoreUtil.newContextException( table.getPos(), RESOURCE.ddlOnSourceTable() );
-        }
-        if ( refTable.entityType != EntityType.ENTITY ) {
-            throw CoreUtil.newContextException( referencesTable.getPos(), RESOURCE.ddlOnSourceTable() );
-        }
-        try {
-            DdlManager.getInstance().addForeignKey(
-                    catalogTable,
-                    refTable,
-                    columnList.getList().stream().map( Node::toString ).collect( Collectors.toList() ),
-                    referencesList.getList().stream().map( Node::toString ).collect( Collectors.toList() ),
-                    constraintName.getSimple(),
-                    onUpdate,
-                    onDelete );
-        } catch ( UnknownColumnException e ) {
-            throw CoreUtil.newContextException( columnList.getPos(), RESOURCE.columnNotFound( e.getColumnName() ) );
-        } catch ( GenericCatalogException e ) {
-            throw new RuntimeException( e );
-        }
+    public void execute( Context context, Statement statement, ParsedQueryContext parsedQueryContext ) {
+        LogicalTable logicalTable = getTableFailOnEmpty( context, table );
```

Comment:
Add a check whether the found schema object actually is a table and not, for instance, a view.

Reply:
Added handling.